src/state.rs

```diff
@@ -57,7 +57,8 @@ where
         let server = self.server().lock().await;
         let ks = self.key_store();
         let ns = self.nonce_store();
-        server.redeem_token(ks, ns, token.clone()).await.is_ok()
+        let res = server.redeem_token(ks, ns, token.clone()).await;
+        res.is_ok()
```

Comment:
Is this change really necessary?

Reply:
Upgrading to the 2024 edition triggered a warning here because of the changed drop order. It would probably have been fine without the change, but this way there is no ambiguity.